vnext/Desktop/Modules/BlobModule.cpp

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "BlobModule.h"
+
+#include <unicode.h>
+
+// React Native
+#include <cxxreact/Instance.h>
+#include <cxxreact/JsArgumentHelpers.h>
+#include <cxxreact/ModuleRegistry.h>
+
+// Windows API
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Security.Cryptography.h>
+
+using namespace facebook::xplat;
+
+using folly::dynamic;
+using std::lock_guard;
+using std::mutex;
+using std::string;
+using std::vector;
+using std::weak_ptr;
+using winrt::Windows::Foundation::GuidHelper;
+using winrt::Windows::Security::Cryptography::CryptographicBuffer;
+
+namespace {
+constexpr char moduleName[] = "BlobModule";
+constexpr char blobURIScheme[] = "blob";
+
+weak_ptr<Microsoft::React::IWebSocketModuleContentHandler> s_contentHandler;
```

Comment:
this wouldn't allow multiple instances using the module?

Reply:
Don't se how. Can you elaborate?

If it's regarding the constructor's initialization, it's been fixed:

```
  if (!m_contentHandler) {
    m_contentHandler = std::make_shared<BlobWebSocketModuleContentHandler>();
    s_contentHandler = m_contentHandler;
  }
```

Reply:
I'm also kind of confused by this. E.g. does `WebSocketModuleContentHandler::GetInstance` return the most recent content handler? Can we reason about the state of this when we call it? Do we keep a static weak_ptr to let the singleton instance go away? Etc.

Would be curious to know why we chose that route. Taking a look at first glance, it's not immediately clear how to reason about what it's doing. It feels like we might be able to achieve the same goals in a clearer way.

Reply:
In this workflow, we only need one handler instance. In the end, all the handler does is fill out folly args and store the blobs' contents.

Here is my line of reasoning (admittedly, should probably have written this in detail in the description):
- There is a state that needs to be tracked, containing:
  - Blob binary contents
  - WebSocket session IDs for those sessions that will use the blob manager.

iOS and Android do this by having the BlobModule directly grab a reference to the WebSocketModule class, and WebSocketModuleModule store multiple references (mapped by session ID) to the BlobModule (still a singleton).

On Windows, Both WS and Blob modules are stored as `unique_ptr`, so they can't be exposed/shared (and shouldn't, in my opinion).
The proposed solution is for one of those two modules to own a `shared_ptr` that keeps the state mentioned above.

The detail is, being more of a "blob-related" functionality than a "websocket-related" one, I chose the blob one to ultimately own this shared state, so it should lose its owning reference when BlobModule is destroyed.

Reply:
Do we keep a static weak_ptr to let the singleton instance go away? Etc.

So, to actually answer the questions:
> does `WebSocketModuleContentHandler::GetInstance` return the most recent content handler?

It returns the ONLY content handler.

> Can we reason about the state of this when we call it?

See comment above.

> Do we keep a static weak_ptr to let the singleton instance go away?

Yes.